### PR TITLE
Commit exo test docker-compose files

### PR DESCRIPTION
**Pull request by hugoesthere**
Partially resolves: https://github.com/Originate/exosphere/issues/697

### CODE REVIEW

File: src/application/tester/test_runner.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+func (s *TestRunner) getDockerComposeConfigs() (types.DockerConfigs, error) {
+	return composebuilder.GetApplicationDockerConfigs(composebuilder.ApplicationOptions{
+		AppConfig: s.AppConfig,
+		AppDir:    s.AppDir,
+		BuildMode: s.BuildMode,
+		HomeDir:   s.HomeDir,
+	})
+}
```
Comment (hugoesthere) [inline src/application/tester/test_runner.go]: This was moved from `src/application/service_tester.go` but also cleaned up significantly

File: src/application/tester/test_runner.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	builtDependencies := config.GetBuiltDevelopmentDependencies(appConfig, serviceConfigs, appDir, homeDir)
+	if err != nil {
+		return nil, err
+	}
+
+	tester := &TestRunner{
+		AppConfig:                appConfig,
+		AppDir:                   appDir,
+		BuildMode:                mode,
+		BuiltDependencies:        builtDependencies,
```
Comment (charlierudolph) [inline src/application/tester/test_runner.go]: Are these used anywhere? If not can we please delete them?

File: src/application/tester/test_runner.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+// NewTestRunner is TestRunner's constructor
+func NewTestRunner(appContext types.AppContext, writer io.Writer, mode composebuilder.BuildMode) (*TestRunner, error) {
+	appConfig := appContext.Config
+	appDir := appContext.Location
+	homeDir, err := util.GetHomeDirectory()
+	if err != nil {
+		return nil, err
+	}
+	dockerComposeProjectName := fmt.Sprintf("%stests", composebuilder.GetDockerComposeProjectName(appConfig.Name))
```
Comment (charlierudolph) [inline src/application/tester/test_runner.go]: This is only used once. Can we create it `getRunOptions`?

File: src/application/tester/test_runner.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+	tester := &TestRunner{
+		AppConfig:                appConfig,
+		AppDir:                   appDir,
```
Comment (charlierudolph) [inline src/application/tester/test_runner.go]: Lets save appContext instead of appConfig and appDir.

File: src/application/tester/tester.go [hunk 5]
```diff
@@ -52,17 +58,13 @@ func TestApp(appContext types.AppContext, writer io.Writer, mode composebuilder.
 
 // TestService runs the tests for the service and return true if the tests passed
 // and an error if any
-func TestService(serviceContext types.ServiceContext, writer io.Writer, mode composebuilder.BuildMode, shutdown chan os.Signal) (types.TestResult, error) {
-	util.PrintSectionHeaderf(writer, "Testing service '%s'\n", serviceContext.Dir)
-	serviceTester, err := NewServiceTester(serviceContext, writer, mode)
-	if err != nil {
-		return types.TestResult{}, err
-	}
+func TestService(testRunner *TestRunner, testRole string, writer io.Writer, shutdown chan os.Signal) (types.TestResult, error) {
```
Comment (charlierudolph) [inline src/application/tester/tester.go]: Can we keep TestService with the api it had and create a private `runServiceTest` with the shared implementation? I want to keep as much logic out of the cmd folder as we can

**Comment by hugoesthere** [comment]
@charlierudolph comments addressed

**Review by charlierudolph — APPROVED** [review]

**Pull request merged**